### DISCUSSION TITLE
Can I use ccache in debug environment only?

### DISCUSSION THREAD

**Discussion by namedssy** [ccache/ccache#317] [Questions & answers] [answered]
Hello, we boss said "Only used ccache in debug environment, not when Archive packaging",so, can i use ccache in debug environment only please？

**Comment by afbjorklund** [comment, marked as answer]
Yes, you can disable ccache for the release builds... Either by *not* prefixing your compiler with `ccache`, or by setting the $CCACHE_DISABLE environment variable if "masquerading".

**Comment by namedssy** [comment]
thank you very much !
![image](https://user-images.githubusercontent.com/16632642/45632294-89eff080-bacf-11e8-86fb-4d0f718b5097.png)
Am I correcting this way?

there is another question I would like to ask
if i want to auto configuration setup $CCACHE_DISABLE environment in release environment,how should i do please?

**Comment by afbjorklund** [comment]
You don't need to do *both* of them, either you add ccache (and don't disable it) or you don't add it... Presumably you have different configuration for "debug environment" and "release environment" ?

Wrapping each and every compiler invocation is probably overkill for this setup, better to set the configuration once (like source a setup file) and then reuse the same environment for all compiles.
